.github/workflows/test-coverage.yaml

```diff
@@ -17,7 +17,7 @@ jobs:
       - name: Setup python
         uses: actions/setup-python@v4
         with:
-          python-version: 3.x
```

Comment:
This line is the main culprit here, as this will now install `3.12` by default.

Version locking this to `3.11` works for now, but we should eventually update this and add `3.12` to the test matrix once wheels are available.